File: solutions/java/rotational-cipher/1/src/main/java/RotationalCipher.java
class RotationalCipher {
    private final int shiftKey;

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {
        StringBuilder result = new StringBuilder();
        
        for (char c : data.toCharArray()) {
            if (c >= 'a' && c <= 'z') {
                // Shift lowercase letters, wrap around with modulo
                result.append((char) ('a' + (c - 'a' + shiftKey) % 26));
            } else if (c >= 'A' && c <= 'Z') {
                // Shift uppercase letters, wrap around with modulo
                result.append((char) ('A' + (c - 'A' + shiftKey) % 26));
            } else {
                // Preserve spaces, punctuation, numbers, etc.
                result.append(c);
            }
        }
        
        return result.toString();
    }
}

File: solutions/javascript/rotational-cipher/1/rotational-cipher.js
export const rotate = (text, key) => {
  let result = '';

  for (let i = 0; i < text.length; i++) {
    const char = text[i];
    const code = text.charCodeAt(i);

    if (code >= 65 && code <= 90) {
      // Uppercase A-Z
      result += String.fromCharCode(((code - 65 + key) % 26) + 65);
    } else if (code >= 97 && code <= 122) {
      // Lowercase a-z
      result += String.fromCharCode(((code - 97 + key) % 26) + 97);
    } else {
      // Non-alphabetic characters pass through unchanged
      result += char;
    }
  }

  return result;
};
